shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/DistSQLBackendHandlerFactory.java

```diff
@@ -51,6 +53,7 @@ public static TextProtocolBackendHandler newInstance(final DatabaseType database
         if (sqlStatement instanceof RQLStatement) {
             return RQLBackendHandlerFactory.newInstance((RQLStatement) sqlStatement, connectionSession);
         }
+        checkLockedSchema(connectionSession);
         if (sqlStatement instanceof RDLStatement) {
```

Comment:
scaling DistSQL should be executable, they're RAL.

But I'm not sure whether other RAL could be executable or not.

Reply:
I will optimize the type of statement intercepted `QueryableRALStatement`, `QueryableScalingRALStatement`, `UpdatableScalingRALStatement` they will be ignored.